client/auth/controller.js

```diff
@@ -25,14 +35,84 @@ module.exports = {
 	},
 
 	checkToken: function( context, next ) {
-		const loggedOutRoutes = [ '/login', '/oauth', '/start' ],
+		const loggedOutRoutes = [ '/login', '/oauth', '/start', '/authorize', '/api/oauth/token' ],
 			isValidSection = loggedOutRoutes.some( route => startsWith( context.path, route ) );
 
-		// Check we have an OAuth token, otherwise redirect to login page
+		if ( config( 'env' ) === 'desktop' ) {
+			// Check we have an OAuth token, otherwise redirect to login page
+			if ( OAuthToken.getToken() === false && ! isValidSection ) {
+				page( '/login' );
+			} else {
+				next();
+			}
+		}
```

Comment:
I'm not familiar with the domain, so a couple of potentially dumb questions:
- Shouldn't we `return` at the end of the `if ( config( 'env' ) === 'desktop' ) {}` block, so that, when running on the Desktop client, we never reach the remaining `page/next` calls?
- What are we checking with `OAuthToken.getToken() === false && ! isValidSection`? Usually, I'd expect a `||` instead of a `&&`.